testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -134,6 +141,131 @@ public void testColumnMappingMode(String mode)
+                " 'delta.minWriterVersion'='5')");
+
+        try {
+            onTrino().executeQuery("" +
+                    "INSERT INTO delta.default." + tableName +
+                    " VALUES " +
+                    "(" +
+                    "   true, " +
+                    "   1, " +
+                    "   10," +
+                    "   100, " +
+                    "   1000, " +
+                    "   CAST('123.12' AS DECIMAL(5,2)), " +
+                    "   CAST('123456789012345678.123' AS DECIMAL(21,3)), " +
+                    "   DOUBLE '0', " +
+                    "   REAL '0', " +
+                    "   'a', " +
+                    "   DATE '2020-08-21', " +
+                    "   TIMESTAMP '2020-10-21 01:00:00.123 UTC', " +
+                    "   X'abcd', " +
+                    "   ARRAY['element 1'], " +
+                    "   ARRAY[ROW('nested 1')], " +
+                    "   MAP(ARRAY['key'], ARRAY['value1']), " +
+                    "   MAP(ARRAY['key'], ARRAY[ROW('nested value1')]), " +
+                    "   ROW('item 1', 1), " +
+                    "   ROW(ROW('nested item 1'), 11) " +
+                    (partitioned ? ", 'part1'" : "") +
+                    "), " +
+                    "(" +
+                    "   true, " +
+                    "   2, " +
+                    "   20," +
+                    "   200, " +
+                    "   2000, " +
+                    "   CAST('223.12' AS DECIMAL(5,2)), " +
+                    "   CAST('223456789012345678.123' AS DECIMAL(21,3)), " +
+                    "   DOUBLE '0', " +
+                    "   REAL '0', " +
+                    "   'b', " +
+                    "   DATE '2020-08-22', " +
+                    "   TIMESTAMP '2020-10-22 02:00:00.456 UTC', " +
+                    "   X'abcd', " +
+                    "   ARRAY['element 2'], " +
+                    "   ARRAY[ROW('nested 2')], " +
+                    "   MAP(ARRAY['key'], ARRAY[null]), " +
+                    "   MAP(ARRAY['key'], ARRAY[null]), " +
+                    "   ROW('item 2', 2), " +
+                    "   ROW(ROW('nested item 2'), 22) " +
+                    (partitioned ? ", 'part2'" : "") +
+                    ")");
+
+            List<Row> expectedRows = ImmutableList.of(
+                    row(true, 1, 10, 100, 1000L, new BigDecimal("123.12"), new BigDecimal("123456789012345678.123"), 0d, 0f, "a", java.sql.Date.valueOf(LocalDate.of(2020, 8, 21)), new byte[] {(byte) 0xAB, (byte) 0xCD}, "element 1", "nested 1", "value1", "nested value1", "item 1", 1, "nested item 1", 11),
+                    row(true, 2, 20, 200, 2000L, new BigDecimal("223.12"), new BigDecimal("223456789012345678.123"), 0d, 0f, "b", java.sql.Date.valueOf(LocalDate.of(2020, 8, 22)), new byte[] {(byte) 0xAB, (byte) 0xCD}, "element 2", "nested 2", null, null, "item 2", 2, "nested item 2", 22));
+
+            String selectDatabricksValues = "SELECT " +
```

Comment:
nit: I would rename to `selectDeltaValues` for compatibility with `onDelta` because it's not specific to Databricks.